### PR TITLE
UGENE-7645 Primer3 creates 2 annotation tables

### PR DESCRIPTION
**Pull request by rasputinkirill**

### CODE REVIEW

File: src/plugins/GUITestBase/src/GTUtilsProjectTreeView.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/GTUtilsProjectTreeView.cpp]: The method is incorrect because project view may be in the filtering state.
You can just rename it to "countTopLevelItems" - in this case it will be good.

File: src/plugins/GUITestBase/src/GTUtilsProjectTreeView.h
Comment (yalgaer) [inline src/plugins/GUITestBase/src/GTUtilsProjectTreeView.h]: Fix the comment. Start with a verb, end with a dot.

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: Use showContextMenu::showContextMenu

File: src/plugins_3rdparty/primer3/src/Primer3Dialog.cpp
Comment (yalgaer) [inline src/plugins_3rdparty/primer3/src/Primer3Dialog.cpp]: Why do you want to close the dialog if the action failed? May be allow user to change some settings and submit the dialog again?

**Review by yalgaer — CHANGES_REQUESTED** [review]

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: Can you check the items by names here too, so it is not random items, but the expected ones?

**Review by DmitriiSukhomlinov — CHANGES_REQUESTED** [review]
Please, do not merge it into master, it makes tons of complications to me, because I will may have a lot merge conflicts. Merge it to `primer3-update`

File: src/plugins_3rdparty/primer3/src/Primer3Dialog.cpp
Comment (rasputinkirill) [reply src/plugins_3rdparty/primer3/src/Primer3Dialog.cpp]: Agreed

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (rasputinkirill) [reply src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: I used this solution because we cant be sure about generated names in default directory for annotations. But we are sure about quantity of files in project after  performing Primer3 execution.

File: src/plugins_3rdparty/primer3/src/Primer3Plugin.cpp
Comment (DmitriiSukhomlinov) [inline src/plugins_3rdparty/primer3/src/Primer3Plugin.cpp:128]: Split to separate lines

**Review by DmitriiSukhomlinov — APPROVED** [review]
Minor comment, but please, wait for @yalgaer approve before merge

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (yalgaer) [reply src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: Every test is run in isolation and tests are reproducible. 
It means that the produced file names in the isolated environment are the same.
Why don't you know the generated file names? If there is a big problem with it check part of the names &types.

**Review by yalgaer — APPROVED** [review]

**Pull request merged**
